meteor/server/migration/addMissingIdsForSelectFromColumns.ts

```diff
@@ -0,0 +1,203 @@
+import { Blueprints } from '../../lib/collections/Blueprints'
+import {
+	ConfigItemValue,
+	ConfigManifestEntry,
+	ConfigManifestEntrySelectFromColumn,
+	ConfigManifestEntryType,
+	IBlueprintConfig,
+	MigrationStepBase,
+	TableConfigItemValue,
+} from '@sofie-automation/blueprints-integration'
+import { ShowStyleBase, ShowStyleBases } from '../../lib/collections/ShowStyleBases'
+import { BlueprintId, ShowStyleVariantId } from '@sofie-automation/corelib/dist/dataModel/Ids'
+import { ShowStyleVariant, ShowStyleVariants } from '../../lib/collections/ShowStyleVariants'
+import { Blueprint } from '@sofie-automation/corelib/dist/dataModel/Blueprint'
+
+interface BlueprintWithShowStyleConfigManifest extends Blueprint {
+	showStyleConfigManifest: ConfigManifestEntry[]
+}
+
+export function createMigrationsForAddingMissingIdsInSelectFromColumnEntries(): MigrationStepBase[] {
+	const blueprints = Blueprints.find({
+		showStyleConfigManifest: { $exists: true },
+	}).map((blueprint) => ({ blueprintId: blueprint._id, showStyleConfigManifest: blueprint.showStyleConfigManifest }))
+
+	const baseMigrations: MigrationStepBase[] = createMigrationsForShowStyleBase(
+		blueprints as BlueprintWithShowStyleConfigManifest[]
+	)
+	const variantMigrations: MigrationStepBase[] = createMigrationsForShowStylVariants(
+		blueprints as BlueprintWithShowStyleConfigManifest[]
+	)
+	return [...baseMigrations, ...variantMigrations]
+}
+
+function createMigrationsForShowStyleBase(blueprints: BlueprintWithShowStyleConfigManifest[]): MigrationStepBase[] {
+	return blueprints
+		.map((blueprint) =>
+			blueprint.showStyleConfigManifest.map((showStyleManifest) =>
+				createShowStyleBlueprintConfigurationSelectFromColumnMigration(blueprint.blueprintId, showStyleManifest)
+			)
+		)
+		.flat()
+}
+
+function createShowStyleBlueprintConfigurationSelectFromColumnMigration(
+	blueprintId: BlueprintId,
+	configManifestEntry: ConfigManifestEntry
+): MigrationStepBase {
+	return {
+		id: `${blueprintId}.add.missing.id.to.select.from.column.for.${configManifestEntry.name}.${configManifestEntry.id}`,
+		canBeRunAutomatically: true,
+		validate: () => {
+			const entryToMapFrom = findEntryToMapFromForShowStyleBase(blueprintId, configManifestEntry)
+			// If we find an entry to map that means we have an id to map and should run the migration
+			return !!entryToMapFrom
+		},
+		migrate: () => {
+			const entryToMapFrom = findEntryToMapFromForShowStyleBase(blueprintId, configManifestEntry)
+			if (!entryToMapFrom) {
+				return
+			}
+			ShowStyleBases.update(
+				{
+					blueprintId,
+				},
+				{
```

Comment:
Correct me if I'm wrong, but since it is possible to have multiple ShowStyleBases reference the same `blueprintId`, won't this overwrite entries in unrelated showstyles?

Reply:
It's been updated